lib/tasks/services.rake

```diff
@@ -12,4 +12,17 @@ namespace :services do
   task :destroy_marked_as_deleted => :environment do
     DestroyAllDeletedObjectsWorker.perform_later(Service.to_s)
   end
+
+  desc 'Destroy a service in the Background as long as the provider has at least 1 more service. Whether it is a default one or not'
+  task :destroy_default_service, %i[account_id service_id] => [:environment] do |_task, args|
+    account = Account.providers.find(args[:account_id])
+    service_id = args[:service_id]
+    service = account.services.find(service_id) # So it will raise an error if this service does not belong to this account :)
+    if service.default?
+      new_default_service = account.services.where.not(id: service_id).select(:id).first! # It will raise an error if it does not have another service
```

Comment:
```suggestion
      new_default_service = account.services.where.not(id: service_id).first! # It will raise an error if it does not have another service
```

`.select(:id)` is ok here because you know `Service#backend_id` only depends on this attribute of the model, but this is breaking the abstraction and we should not limit this way.

Reply:
Right :+1: